packages/design-system/src/components/web-components/preactement/define.ts

```diff
@@ -97,20 +98,26 @@ function setupElement<T>(component: ComponentFunction<T>, options: IOptions = {}
           if (this.__mounted) {
             this.attributeChangedCallback(name, null, v);
           }
-  
+
           const type = typeof v;
-          if (
-            v == null ||
-            type === 'string' ||
-            type === 'boolean' ||
-            type === 'number'
-          ) {
+          if (v == null || type === 'string' || type === 'boolean' || type === 'number') {
             this.setAttribute(name, v);
           }
         },
       });
     });
 
+    // This works, but it clobbers pre-existing event handlers made via `addEventListener`
+    // events.forEach((name) => {
+    //   Object.defineProperty(CustomElement.prototype, name, {
+    //     set(v) {
+    //       if (this.__mounted) {
+    //         this.attributeChangedCallback(name, null, v);
```

Comment:
Just had an idea that we could explore later: What if instead of trying to define the callback that they add as one of the props (which clobbers any listeners we made before) we instead added an event listener in this function? Like `bob.onChange = myCallback` results in `bob.addEventListener('ds-change', myCallback)`?

Reply:
We could pull the `kebabCaseIt(name.replace('on', 'ds'))` logic out into its own function and then use it here in this `set` function and in the `proxyEvents` function